src/core/cache.ts

```diff
@@ -24,6 +26,18 @@ export class Cache {
     this.setCacheControl("no-preview")
   }
 
+  set store(store: string | CacheStore) {
```

Comment:
The cache API is only available in a secure context. It's an edge case for people to be on plain http these days, but just in case, would it be worth checking that the API is available when setting the cache, and raising an error with an explanation if it's not?

Reply:
👍 added in 2d0985c7bc131bfb3a72c40bf0cb7941baf50bd4